lib/blockchain_double.js

```diff
@@ -418,29 +418,35 @@ BlockchainDouble.prototype.sortByPriceAndNonce = function() {
   self.pending_transactions = sorted_transactions;
 };
 
-BlockchainDouble.prototype.processCall = function(tx, callback) {
+BlockchainDouble.prototype.processCall = function(tx, blockNumber, callback) {
   var self = this;
+  var startingStateRoot = self.stateTrie.root;
 
+  var cleanUpAndReturn = function (err, result, changeRoot) {
+    // For defaultBlock, undo state root changes
+    if (changeRoot){
+      self.stateTrie.root = startingStateRoot;
+    }
```

Comment:
No need for the changeRoot conditional here. There's no direct side-effect to the assignment. I think root is a property, but under the hood it's just updating a field to point at whatever buffer object you give it. IIRC, it doesn't even lock (hence trufflesuite/ganache-cli#453

Reply:
In the end this seemed to be necessary and needs to happen after the `vm.revert`. No idea why. Expanded the tests to include a case where we call `latest` immediately after calling a specific block and verify that the response is correct.